libcontainer/cgroups/fs/blkio.go

```diff
@@ -42,11 +42,15 @@ func (s *BlkioGroup) Set(path string, r *configs.Resources) error {
 		}
 	}
 	for _, wd := range r.BlkioWeightDevice {
-		if err := fscommon.WriteFile(path, s.weightDeviceFilename, wd.WeightString()); err != nil {
-			return err
+		if wd.Weight != 0 {
+			if err := fscommon.WriteFile(path, s.weightDeviceFilename, wd.WeightString()); err != nil {
+				return err
+			}
 		}
-		if err := fscommon.WriteFile(path, "blkio.leaf_weight_device", wd.LeafWeightString()); err != nil {
-			return err
+		if wd.LeafWeight != 0 {
```

Comment:
The question is, CFQ (removed from kernel v5.0) supported weight and leaf_weight, while BFQ only supports weight. Should we ignore leaf weight for BFQ case or error out?

Reply:
I think erroring out is fine, as leaf weights are obsoleted and no one should set them. If they do, they get an error.